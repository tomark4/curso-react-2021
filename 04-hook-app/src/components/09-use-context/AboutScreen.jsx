import { useContext } from "react"
import {UserContext} from './UserContextProvider'

const AboutScreen = () => {
    const {user} = useContext(UserContext);
    return (
        <>
            <h1>About</h1>
            <pre>
                {JSON.stringify(user)}
            </pre>
        </>
    )
}

export default AboutScreen
