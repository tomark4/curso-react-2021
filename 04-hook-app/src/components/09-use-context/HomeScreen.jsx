import { useContext } from "react"
import {UserContext} from './UserContextProvider'

const HomeScreen = () => {
    const {user} = useContext(UserContext);

    return (
        <div>
            <h1>Home</h1>
            <pre>
                {JSON.stringify(user)}
            </pre>
        </div>
    )
}

export default HomeScreen
