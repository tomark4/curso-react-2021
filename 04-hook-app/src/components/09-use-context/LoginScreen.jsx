import { useContext } from "react"
import {UserContext} from "./UserContextProvider"

const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    return (
        <>
            <h1>Login</h1>
            <button className="btn btn-primary"
            onClick={() => setUser({id: 1, name:'jose',email:'jotta@jotta.com'})}>Ingresar</button>

            <button className="btn btn-danger"
            onClick={() => setUser({})}>
                Logout
            </button>
        </>
    )
}

export default LoginScreen
