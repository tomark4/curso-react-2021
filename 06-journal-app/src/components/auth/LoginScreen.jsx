import { useDispatch, useSelector } from 'react-redux';
import { useState } from 'react';
import { Link } from 'react-router-dom';
import { loginWithGoogle, loginWithEmailPassword } from '../../actions/auth';
import validator from 'validator';
import {
    removeError,
    setError
} from '../../actions/ui';

const LoginScreen = () => {
    const dispatch = useDispatch();
    const { ui } = useSelector( state => state );

    const [email, setEmail] = useState("test1@jotta.com");
    const [password, setPassword] = useState("123456");

    const handleSubmit = (e) => {
        e.preventDefault();
        if( !validator.isEmail(email) ){
            dispatch( setError('email invalid') );
            return false;
        }
        if( validator.isEmpty(password) || password.length < 6){
            dispatch( setError('password invalid') );
            return false;
        }
        dispatch( removeError() );
        dispatch( loginWithEmailPassword(email, password) );
    }

    const handleLoginGoogle = () => {
        dispatch( loginWithGoogle() );
    }

    return (
        <>
            <h3 className='auth__title'>Ingresar</h3>
            <form onSubmit={handleSubmit}>
                { ui?.errorMessage &&
                    <div className="auth__alert-error">{ ui?.errorMessage }</div>
                }
                <div>
                    <input
                    type="email"
                    placeholder="Email"
                    name="email"
                    className="auth__input"
                    autoComplete="off"
                    value={email}
                    onChange={(e) => setEmail(e.target.value)}/>
                </div>

                <div>
                    <input
                    type="password"
                    placeholder="Password"
                    name="password"
                    className="auth__input"
                    value={password}
                    onChange={(e) => setPassword(e.target.value)}/>
                </div>

                <div>
                    <button type="submit" className={`btn btn-primary btn-block btn-lg ${ui.loading && 'disabled'}`}
                    disabled={ui.loading}>
                        {ui.loading ? (
                            <>
                                <i class="fa fa-spinner fa-spin"></i> Validando
                            </>
                        ) : 'Ingresar'}
                    </button>
                </div>

                <div className="google-btn" onClick={handleLoginGoogle}>
                    <div className="google-icon-wrapper">
                        <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                    </div>
                    <p className="btn-text">
                        <b>Sign in with google</b>
                    </p>
                </div>

            </form>
            <Link to="/auth/register" className="link">Â¿No tienes cuenta? Registrate</Link>
        </>
    )
}

export default LoginScreen
