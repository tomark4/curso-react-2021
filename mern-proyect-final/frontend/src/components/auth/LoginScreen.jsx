import './login.css';

const LoginScreen = () => {

    return (
        <div className="container">
            <div className="row">
                <div className="col s12 m6">
                    <div className="row mt-5" style={{padding:"40px"}}>
                        <h3 className="mb-5">Login</h3>
                        <form autoComplete="off">
                            <div className="input-field">
                                <input type="text" name="email" placeholder="Email"
                                id="email"/>
                            </div>
                            <div className="input-fiele">
                                <input type="password" name="password" placeholder="Password"
                                id="password"/>
                            </div>
                            <div className="input-field">
                                <button type="submit" className="btn indigo accent-3"
                                style={{ width:"100%"}}>Login</button>
                            </div>
                        </form>
                    </div>
                </div>
                <div className="col s12 m6">
                    <div className="row" style={{padding:"40px"}}>
                        <h3 className="mb-5">Register</h3>
                        <form className="col s12" autoComplete="off">
                            <div className="row">
                                <div className="input-field col s12 m6">
                                    <input placeholder="First name" id="first_name" type="text" className="validate" />
                                </div>
                                <div className="input-field col s12 m6">
                                    <input id="last_name" type="text" placeholder="Last name"/>
                                </div>
                            </div>
                            <div className="row">
                                <div className="input-field col s12">
                                    <input id="email" type="email" placeholder="Email"/>
                                </div>
                            </div>
                            <div className="row">
                                <div className="input-field col s12">
                                    <input id="password" type="password" placeholder="Password"/>
                                </div>
                            </div>
                            <div className="row">
                                <button
                                className="btn indigo accent-3"
                                type="submit"
                                style={{width:"100%"}}
                                name="action">Register
                                </button>
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default LoginScreen
