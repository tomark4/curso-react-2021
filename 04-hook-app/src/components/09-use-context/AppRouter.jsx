import {
    BrowserRouter as Router,
    Switch,
    Route
} from 'react-router-dom';

import AboutScreen from "./AboutScreen";
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import NavBar from './NavBar';
import NotFoundPage from './NotFoundScreen';

const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Switch>
                    <Route path="/" component={HomeScreen} exact></Route>
                    <Route path="/login" component={LoginScreen} exact></Route>
                    <Route path="/about" component={AboutScreen} exact></Route>
                    <Route component={ NotFoundPage } exact></Route>
                </Switch>
            </div>
        </Router>
    )
}

export default AppRouter
