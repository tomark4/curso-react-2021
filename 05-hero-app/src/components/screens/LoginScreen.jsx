import { useContext } from "react";
import {AuthContext} from '../../auth/AuthContextProvider';
import {types} from '../../types/types';

const LoginScreen = ({history}) => {
    const {dispatch} = useContext(AuthContext);

    const lastPath = localStorage.getItem('lastPath') || "/";

    const handleLogin = () => {
        dispatch({
            type: types.login,
            payload: {
                name: 'jose'
            }
        });
        history.replace(lastPath)
    }

    return (
        <div>
            <div className="container mt-5">
                <div className="row">
                    <div className="col-6 mx-auto">
                        <div className="card">
                            <div className="card-body">
                                <h4>Login</h4>
                                <hr/>
                                <button className="btn btn-success btn-block"
                                onClick={handleLogin}>Login</button>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default LoginScreen
