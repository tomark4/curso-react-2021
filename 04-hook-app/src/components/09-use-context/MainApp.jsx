import AppRouter from './AppRouter';
const MainApp = () => {
    return (
        <>
         <AppRouter />
        </>
    )
}

export default MainApp
