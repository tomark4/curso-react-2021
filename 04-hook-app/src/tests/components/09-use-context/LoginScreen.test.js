import { mount } from "enzyme";
import LoginScreen from "../../../components/09-use-context/LoginScreen";
import {UserContext} from "../../../components/09-use-context/UserContextProvider";

describe('Pruebas en en login screen component', () => {
  test('debe mostrarse correctamente', () => {
    const wrapper = mount(<UserContext.Provider value={{
      setUser: jest.fn(),
      user: {}
    }}>
      <LoginScreen />
    </UserContext.Provider>);

    expect(wrapper).toMatchSnapshot();
  })


  const setUser = jest.fn()
  const wrapper = mount(
    <UserContext.Provider value={{setUser}}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('should ejecutar el setUser con el argumento esperado', () => {
    wrapper.find('.btn-primary').simulate('click');
    expect(setUser).toHaveBeenCalledWith({id: 1, name:'jose',email:'jotta@jotta.com'})
  });

})
