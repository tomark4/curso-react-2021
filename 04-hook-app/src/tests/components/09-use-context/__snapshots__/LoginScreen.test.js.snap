// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en en login screen component debe mostrarse correctamente 1`] = `
Array [
  <h1>
    Login
  </h1>,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Ingresar
  </button>,
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Logout
  </button>,
]
`;
